ts/src/transports/http/xhr.ts

```diff
@@ -80,12 +91,35 @@ class XHR implements Transport {
     });
   }
 
+  protected configureXhr(): void {
+    this.xhr.responseType = "text";
+
+    // overriding the mime type causes a response that has a code point per byte, which can be decoded using the
+    // stringToArrayBuffer function.
+    this.xhr.overrideMimeType("text/plain; charset=x-user-defined");
+  }
+
   cancel() {
     this.options.debug && debug("XHR.abort");
     this.xhr.abort();
   }
 }
 
+export class MozChunkedArrayBufferXHR extends XHR {
```

Comment:
Removed a tonne of duplicated code between the xhr and mozXhr files by using a classic Template Design Pattern.